bin/specextract

```diff
@@ -828,860 +753,1086 @@ def clip_wmap(wmapfile,threshold,tmpdir):
+    iteminfostr = args["iteminfostr"]
+    pars = args["pars_specextract"]
 
-    return ra,dec,skyx,skyy,chipx,chipy,chip_id
+    weight = args["weight"]
+    weight_rmf = args["weight_rmf"]
 
+    tdetwmap = args["tdetwmap"]
 
-###############################################################
-# Usage:
-#  event_stats(file, colname)
-#
-#
-# Aim:
-#     Use dmstat to determine the event statistics: source chipx,
-#     chipy, sky x, sky y, and ccd_id values to use for input
-#     to asphist, acis_fef_lookup, and arfcorr.
-#     Since dmstat doesn't return the mode, do this by brute force
-#     for ccd_id column (psextract algorithm uses mean ccd_id value,
-#     which isn't appropriate for this quantity).
-#
-################################################################
+    try:
+        bpix = args["bpix"]
+    except KeyError:
+        pass
 
-def event_stats(file, colname):
+    try:
+        da = args["da"]
+    except KeyError:
+        pass
 
-    dmstat.punlearn()
-    dmstat.verbose = "0"
+    try:
+        msk = args["msk"]
+    except KeyError:
+        pass
+       
+    if all([weight == "no", correct == "yes", srcbkg == "src"]):
+        outreg = tempfile.NamedTemporaryFile(suffix=f"_phys_coords_{key_id}.reg",dir=tmpdir)
+        fullfile = convert_region(fullfile,filename,outreg.name,"yes",verbose)
 
-    cr = pcr.read_file(file)
-    if cr is None:
-        raise IOError(f"Unable to read from file {file}")
-    
-    if colname == "ccd_id":
+        
+    ###########################
+    #
+    # create ACIS ARF
+    #
+    ###########################
+
+    with new_pfiles_environment(ardlib=True,copyuser=False):
+        if dobpix:
+            set_badpix(filename,bpix,instrument,verbose)
+
+        infostr = f"Creating {srcbkg} ARF {iteminfostr}"
+        
+        try:
+            if any([srcbkg == "src" and weight == "yes", srcbkg == "bkg" and dobkgresp]):
+                weight = "yes" # force background ARF to always be weighted
+                
+                ancrfile, weightfile = create_arf_ext(full_outroot,
+                                                      ebin,verbose,
+                                                      da,msk,
+                                                      tdetwmap,tmpdir,infostr)
+                weightfile.close()
+                
+                add_tool_history(ancrfile,toolname,pars,toolversion=__revision__)
+
+            if all([srcbkg == "src", weight == "no"]):
+                try:
+                    ancrfile = create_arf_ps(full_outroot,filename,
+                                             asphist,ebin,clobber,verbose,
+                                             da,msk,
+                                             ccd_id,skyx,skyy,infostr)
+
+                    add_tool_history(ancrfile,toolname,pars,toolversion=__revision__)
+                    
+                except OSError:
+                    raise IOError(f"Failed to create ARF for {fullfile}")
```

Comment:
For these errors (there's a number of them here) we should use either

```
except OSError as exc:
    raise OSError(...blah....) from exc
```

or

```
except OSError:
    raise OSError(... blah ...) from None
```

depending on whether we care about the original error or not (it can be useful when using --debug in our scripts to either b able to see the underlying error or not to have to read the error). I'd suggest going for the first version [i.e. in this case you could get an error either from `create_arf_ps` or `add_tool_history` so you may want to be able to find the original error] and then setting them to `raise .. from None` once you've decided that we don't care about the original error.